pkg/provider/ip/provider.go

```diff
@@ -120,11 +154,52 @@ func (p *ipv4Provider) DeInitResource(instance ec2.EC2Instance) error {
 
 // UpdateResourceCapacity updates the resource capacity based on the type of instance
 func (p *ipv4Provider) UpdateResourceCapacity(instance ec2.EC2Instance) error {
+	resourceProviderAndPool, isPresent := p.getInstanceProviderAndPool(instance.Name())
+	if !isPresent {
+		p.log.Error(nil, "cannot find the instance provider and pool form the cache", "node-name", instance.Name())
+		return nil
+	}
+
+	resourceProviderAndPool.lock.Lock()
```

Comment:
Does this lock guard `isPrevPDEnabled`?

Reply:
> Does this lock guard `isPrevPDEnabled`?

Yes, this lock guards `isPrevPDEnabled` and `capacity`.